src/ORM/Table.php

```diff
@@ -874,11 +874,40 @@ public function hasBehavior($name)
     /**
      * Returns an association object configured for the specified alias if any
      *
+     * @deprecated 3.6.0 Use getAssociation() instead.
      * @param string $name the alias used for the association.
      * @return \Cake\ORM\Association|null Either the association or null.
      */
     public function association($name)
     {
+        return $this->getAssociation($name);
+    }
+
+    /**
+     * Returns an association object configured for the specified alias if any.
+     *
+     * The name argument also supports dot syntax to access deeper associations.
+     *
+     * ```
+     * $users = $this->getAssociation('Articles.Comments.Users');
+     * ```
+     *
+     * @param string $name the alias used for the association.
+     * @return \Cake\ORM\Association|null Either the association or null.
+     */
+    public function getAssociation($name)
+    {
+        if (strpos($name, '.') !== false) {
```

Comment:
We normally follow "return early" paradigm so this should instead check for false and return the association.

Reply:
No problem. I'll take care of it now.